platform/windows/display_server_windows.cpp

```diff
@@ -2211,10 +2221,30 @@ LRESULT DisplayServerWindows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARA
 		}
 	}
 
-	// Window doesn't exist or creation in progress, don't handle messages yet.
-	if (!window_created) {
-		window_id = window_id_counter;
-		ERR_FAIL_COND_V(!windows.has(window_id), 0);
+	// WARNING: we get called with events before the window is registered in our collection
+	// specifically, even the call to CreateWindowEx already calls here while still on the stack,
+	// so there is no way to store the  window handle in our collection before we get here
+	// instead, we tunnel an unowned reference to our window context (WindowData) through the
+	// first possible message (WM_NCCREATE) to fix up our window context collection
+	if (INVALID_WINDOW_ID == window_id) {
+		switch (uMsg) {
+			case WM_NCCREATE: {
+				// we receive this during CreateWindowEx, and it carries the context we need
+				CREATESTRUCTW *pCreate = (CREATESTRUCTW *)lParam;
+				WindowData *pWindowData = reinterpret_cast<WindowData *>(pCreate->lpCreateParams);
+
+				// fix this up so we can recognize the remaining messages
+				pWindowData->hWnd = hWnd;
+				window_created = true;
```

Comment:
What you say is correct.  My intention was that the "main" big switch statement below could process WM_NCCREATE if it wanted to in the future.  Before my changes, NCCREATE and around 6-7 other messages related to initial positioning of the window were being discarded.  I tried to make it so that everything CAN get processed, except for WM_GETMINMAXINFO, because not having a valid window handle below is asking for trouble.      

If you prefer I can just DefWindowProcW(...) instead for sure.   Actually, should I always punt to the user proc if there is one?   I didn't even notice that code at the bottom before.

Something like this used in a gross :) fall through switch/case?
```
			if (user_proc) {
				return CallWindowProcW(user_proc, hWnd, uMsg, wParam, lParam);
			}
			return DefWindowProcW(hWnd, uMsg, wParam, lParam);
```

Reply:
good thing you pointed this out, because I actually didn't resolve window_id at all, I just initialized the window struct but didn't then look for window_id.  So any code below would have failed.  Somehow lost that bit in moving the code around.  New version will be better.

Reply:
I reorganized the code and split out the special early handling code into a method, so now the way it is called makes it much clearer that we never continue and always return.  makes it much harder to mess it up later